administrator/components/com_templates/Model/TemplateModel.php

```diff
@@ -148,6 +152,151 @@ public function getDirectoryTree($dir)
 		return $result;
 	}
 
+	/**
+	 * Method to get the core file of override file
+	 *
+	 * @param   string   $file  Override file
+	 * @param   integer  $client Client Id
+	 *
+	 * @return  string  $corefile The full path and file name for the target file, or boolean false if the file is not found in any of the paths.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function loadCoreFile($file, $client_id)
+	{
+		$app           = Factory::getApplication();
+		$filePath      = Path::clean($file);
+		$explodeArray  = explode(DIRECTORY_SEPARATOR, $filePath);
+
+		// Only allow html/ folder
+		if($explodeArray['1'] !== 'html')
+		{
+			$app->enqueueMessage(Text::_('COM_TEMPLATES_ERROR_ONLY_HTML_FOLDER_ALLOWED'), 'error');
+
+			return false;
+		}
+
+		$fileName = basename($filePath);
+		$type     = $explodeArray['2'];
+		$client   = ApplicationHelper::getClientInfo($client_id);
+
+		$componentPath = Path::clean($client->path . '/components/');
```

Comment:
Can we set the $componentPath only if type is component and the $layoutPath only if type is layout ...?

Reply:
Yes, we can. Should i do ?

Reply:
$componentPath is used two section  in </code>Jlayout</code> and </code>Component</code>
So, I think it's fine.